src/find_boundaries.jl

```diff
@@ -0,0 +1,303 @@
+"""
+    find_boundaries!(img::AbstractArray, [region,]; background = 0)
+
+Finds the boundaries that are just within each object, replacing the original image.
+`background` is the scalar value of the background pixels which will not be marked as boundaries.
+`region` indicates which dimensions to detect boundaries along.
+
+See `find_boundaries` for examples.
+"""
+function find_boundaries!(img::AbstractArray{Bool,N};
+                         background=false,
+                         kwargs...
+                        ) where N
+    # Find regions where there is at least one background pixel
+    # centered on a foreground pixel
+    if(background != 0)
+        # background_img = img
+        # foreground_img = .!img
+        img .= .!img .& extremefilt!(copy(img), |; kwargs...)
+        # Alternatively, with ImageFiltering.jl
+        # img .= .!img .& mapwindow(any, img, (3,3))
+    else
+        # background_img = .!img
+        # foreground_img = img
+        img .&= extremefilt!(.!img, |; kwargs...)
+        # Alternatively, with ImageFiltering.jl
+        # img .&= mapwindow(any, .!img, (3,3))
+    end
+    return img
+end
+function find_boundaries!(img::AbstractArray{T};
+                          background = zero(T),
+                          kwargs...) where T
+    # Find regions where that are not homogeneous (all equal)
+    # centered on a foreground pixel
+    # background_img = img .== background
+    # foreground_img = img .!= background
+    allequal(x,y) = ifelse(x == y, x, background)
```

Comment:
Is this better than ternary syntax in this case?

Reply:
I switched to ternary for now